services/API-service/src/api/eap-actions/eap-actions.service.ts

```diff
@@ -39,10 +39,22 @@ export class EapActionsService {
       throw new HttpException({ errors }, HttpStatus.NOT_FOUND);
     }
 
+    const query = `select
+        id
+      from
+        "IBF-pipeline-output".event_pcode
+      where
+        closed = false
+        and pcode = $1`;
+
+    const eventPlaceCodeId = await this.manager.query(query, [
+      eapAction.pcode,
+    ])[0]['id'];
+
     const action = new EapActionStatusEntity();
     action.status = eapAction.status;
     action.pcode = eapAction.pcode;
-    action.event = eapAction.event;
+    action.event = eventPlaceCodeId;
```

Comment:
`eventPlacecodeId` to `eventPlaceCode`

unless there is a difference between the `Place Code Id` and `Place Code` in which case use `eventPlaceCodeId` instead of `eventPlacecodeId`